File: 2810-CollectingChocolates/2810-CollectingChocolates.py
# Last updated: 7/1/2025, 11:04:02 PM
class Solution:
    def rotate_left_by_one(self, arr: List[int]):
        first = arr[0]
        for i in range(1, len(arr)):
            arr[i - 1] = arr[i]
        arr[-1] = first

    def minCost(self, nums: List[int], x: int) -> int:
        min_cost = nums[:]
        ans = float('inf')
        sum_val = 0

        for i in range(len(nums)):
            sum_val = 0
            if i != 0:
                self.rotate_left_by_one(nums)
            for j in range(len(nums)):
                min_cost[j] = min(nums[j], min_cost[j])
                sum_val += min_cost[j]
            ans = min(ans, sum_val + (i * x))

        return ans

File: 2810-CollectingChocolates/2810-CollectingChocolates.java
// Last updated: 7/1/2025, 11:04:06 PM
class Solution {

public long minCost(int[] nums, int x) {
        int n = nums.length;
        int[] min_cost = new int[n];
        System.arraycopy(nums, 0, min_cost, 0, n);
        long ans = Long.MAX_VALUE;
        
        for (int i = 0; i < n; i++) {
            long sum = 0;
            for (int j = 0; j < n; j++) {
                int rotatedIndex = (j + i) % n;
                min_cost[j] = Math.min(nums[rotatedIndex], min_cost[j]);
                sum += min_cost[j];
            }
            ans = Math.min(ans, sum + (long) i * x);
        }
        return ans;
    }
}